### PR TITLE
feat: [CDS-49176]: CD Container step

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
#### Screenshots

https://user-images.githubusercontent.com/98508399/211137717-e5789061-d2d2-43ce-b31d-d07ccad1cc4e.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/types.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  MultiTypeListType,
+  MultiTypeListUIType,
+  MultiTypeSelectOption
+} from '@pipeline/components/PipelineSteps/Steps/StepsTypes'
+import type { StepElementConfig } from 'services/cd-ng'
+import type { ContainerStepInfo, VariableMergeServiceResponse } from 'services/pipeline-ng'
+
+export interface ContainerStepData extends StepElementConfig {
+  spec: Omit<ContainerStepInfo, 'outputVariables' | 'shell'> & {
+    outputVariables?: MultiTypeListUIType | MultiTypeListType | Array<{ id?: string; value?: string; name?: string }>
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/ContainerStep/types.ts:20]: Is id, value & name needed to be optional here? If outputVariables is of type Array, array element should have all the values.

**Review by pankajsingh-harness — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/helper.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import type { UseStringsReturn } from 'framework/strings'
+import { cpuLimitRegex, memorLimityRegex } from '@common/utils/StringUtils'
+import type { MapType, MultiTypeListUIType, SelectOption } from '@pipeline/components/PipelineSteps/Steps/StepsTypes'
+import { getNameAndIdentifierSchema } from '@pipeline/components/PipelineSteps/Steps/StepsValidateUtils'
+import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
+import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'
+import type { ContainerStepData } from './types'
+import { getConnectorSchema, getNameSpaceSchema } from '../PipelineStepsUtil'
+
+export const GetShellOptions = (getString: UseStringsReturn['getString']): SelectOption[] => [
```
[the remark this answers is not in the document]
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/ContainerStep/helper.ts:20]: yes, i knew it. 
Will put it in common place when we have second cut for this step.

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/helper.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            name: listValue.value
+          }))
+
+  return {
+    ...values,
+    spec: {
+      ...values.spec,
+      connectorRef: values.spec.connectorRef,
+      outputVariables: outputVarlist,
+      envVariables: typeof envVar === 'string' ? envVar : map
```
[the remark this answers is not in the document]
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/ContainerStep/helper.ts:78]: sure, will update in next PR

File: src/modules/75-cd/components/PipelineSteps/ContainerStep/types.ts (hunk 1 above)
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/ContainerStep/types.ts:20]: Yes, these can be optional as yaml it is defined as a list with name as key.
For name i will update it as required in next PR.

**Pull request merged**